src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,68 @@
+package duke;
+
+import duke.command.Parser;
+import duke.command.Storage;
+import duke.command.TaskList;
+import duke.command.Ui;
+
+import java.text.ParseException;
+
+/**
+ * The main driver of Duke program.
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
+
+    /**
+     * Constructor for Duke.
+     *
+     * @param filePath a string which stores the directory of the data to be loaded
+     */
+    private Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+        parser = new Parser(tasks, ui);
+    }
+
+    /**
+     * Run the program using CLI.
+     */
+    private void run() {
+        ui.printWelcome();
+
+        // process input
+        String text = ui.readLine();
+        while (!text.equals("bye")) {
+            try {
+                parser.processLine(text); // add, delete, etc
+            } catch (DukeException e) {
+                System.err.println(e.getMessage());
+            } catch (NumberFormatException e) {
```

Comment:
Would it be better to wrap all the exceptions at the source with a `DukeException`, so that the catch block only needs to be concerned with one particular type only?

Reply:
Oh! Thanks for the suggestion. I was wondering why we needed `DukeException` class and could not figure out its usage.